src/shuffle.cc

```diff
@@ -31,8 +32,9 @@ size_t ShuffleChain::LenURIs() {
 /* ensure that our window is as full as it can possibly be. */
 void ShuffleChain::FillWindow() {
     while (_window.size() <= _max_window && _pool.size() > 0) {
+        std::uniform_int_distribution<size_t> rnd_dist{0, _pool.size() - 1};
```

Comment:
I pondered on this one quite some time, then considered it ok, 
because size_t is defined as something like "typedef to the biggest unsigned integer being able to handle all sizes you'll ever encounter"
and the following reasoning:

Question is, does The Standard enforce one of the allowed types as "implementation of size_t"?
I did not find any clear answer, backed by The Standard, which either said "this is ok" or "this is not ok".
But since C++11 size_t has to be at least 16bit, so it wont ever typedef'd to a char and all other interers are allowed as IntType.
So size_t must be implemented as a typedef to one of the allowed primitive integer type, and should therefore be ok.


Anyway:
The biggest value we can reach depends on the type returned by size(), which is size_t.
The problem with this is: It may vary depending on the machine it'll run on.

I'll change this to `unsigned long long` for now, which should be big enough to handle any `size_t` anywhere.
Unsigned because the Google styleguide says "try not to mix signedness", and size() will definitely give an unsigned integer.
(Also: clang will complain about narrowing even with `long long`, so definitely won't do that :sunglasses: )
Don't know if and how this will affect non 64bit systems' performance, though.

Reply:
Thanks for the explanation. It's reasonable to me, so we can keep it.

> Don't know if and how this will affect non 64bit systems' performance, though.

This is part of the "critical path" but we're talking 100s of ms here, so the performance shouldn't really be an issue.